Source/WebCore/rendering/RenderTheme.cpp

```diff
@@ -1418,37 +1311,122 @@ bool RenderTheme::hasListButtonPressed(const RenderObject& renderer) const
+            style.setBorderBottomWidth(borderBox.bottom().value());
+        else
+            style.resetBorderBottom();
+    }
+    if (borderBox.left().value() != static_cast<int>(style.borderLeftWidth())) {
+        style.setBorderLeftWidth(borderBox.left().value());
+        if (borderBox.left().value())
+            style.setBorderLeftWidth(borderBox.left().value());
+        else
+            style.resetBorderLeft();
+    }
+
+    // Padding
+    LengthBox paddingBox = Theme::singleton().controlPadding(appearance, style.fontCascade(), style.paddingBox(), style.effectiveZoom());
+    if (paddingBox != style.paddingBox())
+        style.setPaddingBox(WTFMove(paddingBox));
+
+    // Whitespace
+    if (Theme::singleton().controlRequiresPreWhiteSpace(appearance)) {
+        style.setWhiteSpaceCollapse(WhiteSpaceCollapse::Preserve);
+        style.setTextWrapMode(TextWrapMode::NoWrap);
+    }
+
+    // Width / Height
+    // The width and height here are affected by the zoom.
+    // FIXME: Check is flawed, since it doesn't take min-width/max-width into account.
+    LengthSize controlSize = Theme::singleton().controlSize(appearance, style.fontCascade(), { style.width(), style.height() }, style.effectiveZoom());
+    if (controlSize.width != style.width())
+        style.setWidth(WTFMove(controlSize.width));
+    if (controlSize.height != style.height())
+        style.setHeight(WTFMove(controlSize.height));
+
+    // Min-Width / Min-Height
+    LengthSize minControlSize = Theme::singleton().minimumControlSize(appearance, style.fontCascade(), { style.minWidth(), style.minHeight() }, { style.width(), style.height() }, style.effectiveZoom());
+    if (minControlSize.width.value() > style.minWidth().value())
+        style.setMinWidth(WTFMove(minControlSize.width));
+    if (minControlSize.height.value() > style.minHeight().value())
+        style.setMinHeight(WTFMove(minControlSize.height));
+
+    // Font
+    if (auto themeFont = Theme::singleton().controlFont(appearance, style.fontCascade(), style.effectiveZoom())) {
+        // If overriding the specified font with the theme font, also override the line height with the standard line height.
+        style.setLineHeight(RenderStyle::initialLineHeight());
+        if (style.setFontDescription(WTFMove(themeFont.value())))
+            style.fontCascade().update(nullptr);
+    }
+
+    // Special style that tells enabled default buttons in active windows to use the ActiveButtonText color.
+    // The active window part of the test has to be done at paint time since it's not triggered by a style change.
+    style.setInsideDefaultButton(appearance == StyleAppearance::DefaultButton && element && !element->isDisabledFormControl());
+}
 
-void RenderTheme::adjustCheckboxStyle(RenderStyle& style, const Element*) const
+void RenderTheme::adjustCheckboxStyle(RenderStyle& style, const Element* element) const
 {
-    style.resetPadding();
-    style.resetBorder();
-    style.setBoxShadow(nullptr);
```

Comment:
Isn't this removal a behavior change?

Reply:
No, these are only enabled on iOS today and iOS overrides them. I missed this when I last looked at these methods.